frontend/public/components/utils/webhooks.tsx

```diff
@@ -55,7 +55,42 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
     namespace,
   });
   const tableColumnClasses = getTableColumnClasses(canGetSecret);
-  const webhooks = _.filter(triggers, (trigger) => webhookTriggers.has(trigger.type));
+  const [webhookSecrets, setWebhookSecrets] = React.useState<K8sResourceKind[]>([]);
+  const [secretErrors, setSecretErrors] = React.useState<string[]>([]);
+  const [isLoaded, setLoaded] = React.useState(false);
+
+  const webhooks: WebhookTrigger[] = _.filter(triggers, ({ type }) => webhookTriggers.has(type));
+  const webhookNames: string[] = webhooks.map(
+    (webhook: WebhookTrigger): string => {
+      const triggerProperty = getTriggerProperty(webhook);
+      return _.get(webhook, [triggerProperty, 'secretReference', 'name']);
```

Comment:
Won't this try to fetch secrets for webhooks that don't have a secret reference? You should probably use `reduce` to skip webhooks without a secret reference. Consider `_.uniq` in case multiple webhooks reference the same secret.

I'd also call the var `secretNames` to make it more clear it's the name of the secret itself.

```ts
const secretNames: Set<string> = _.uniq(webhooks.reduce((acc: string[], webhook: WebhookTrigger): string[] => {
  const triggerProperty = getTriggerProperty(webhook);
  const name = _.get(webhook, [triggerProperty, 'secretReference', 'name']);
  return name ? [...acc, name] : acc;
}, []));
```